disnake/app_commands.py

```diff
@@ -346,8 +357,17 @@ def __repr__(self) -> str:
 
     @classmethod
     def from_dict(cls, data: Dict[str, Any]):
-        if data.pop("type", 0) == ApplicationCommandType.user.value:
-            return UserCommand(**data)
+        if data.pop("type", 0) != ApplicationCommandType.user.value:
+            raise ValueError("Invalid payload")
+
+        return UserCommand(
+            name=data["name"],
+            default_permission=data.get("default_permission", True),
+            id=_get_as_snowflake(data, "id"),
+            application_id=_get_as_snowflake(data, "application_id"),
+            guild_id=_get_as_snowflake(data, "guild_id"),
+            version=_get_as_snowflake(data, "version"),
```

Comment:
```suggestion
            id=data.get("id"),
            application_id=data.get("application_id"),
            guild_id=data.get("guild_id"),
            version=data.get("version"),
```
`ApplicationCommand.__init__` already converts these, a `.get` should suffice
(same thing further down)